onmt/translate/Translator.py

```diff
@@ -23,39 +31,115 @@ class Translator(object):
        cuda (bool): use cuda
        beam_trace (bool): trace beam search for debugging
     """
-    def __init__(self, model, fields,
-                 beam_size, n_best=1,
-                 max_length=100,
-                 global_scorer=None,
-                 copy_attn=False,
-                 cuda=False,
-                 beam_trace=False,
-                 min_length=0,
-                 stepwise_penalty=False,
-                 block_ngram_repeat=0,
-                 ignore_when_blocking=[]):
+    def __init__(self, opt, report_score=True, out_file=None):
```

Comment:
This code can't be inside the class. You should be able to make a Translator object without knowing about Opt at all.

Reply:
Make sense, i'll do it.

Reply:
The problem I see with this is that it forces us to add some parameters. For simplicity such parameters must be optional and therefore requires default values, which duplicate default values set in `opts`. It's no big deal but I can become less clear to people if default in `opts` and `Translator` aren't the same.